invenio/ext/jinja2hacks.py

```diff
@@ -90,25 +95,25 @@ def setup_app(app):
     jinja2.environment.Markup = Markup
 
     # Escape/Markup replacement in MarkupSafe library.
-    ## FIXME causes recursive calls in `Markup.__new__` and `escape`
-    #try:
+    # FIXME causes recursive calls in `Markup.__new__` and `escape`
+    # try:
     #    import markupsafe
     #    markupsafe.escape = utf8escape
     #    #markupsafe.Markup = Markup
-    #except ImportError:
+    # except ImportError:
     #    pass
 
     return app
 
 
 def utf8escape(s):
-    """
-    UTF8-8-bit-string-friendly replacement function for MarkupSafe/Jinja2
-    escape function.
+    """UTF8-8-bit-string-friendly replacement for MarkupSafe escape function.
 
     WARNING: Do not use this method. Use jinja2.escape() instead.
     """
     if isinstance(s, str):
+        warnings.warn("Convert string '{0}' in template to unicode.".format(s),
```

Comment:
Perhaps we try to decode using ascii first, and if it fails, then issue the warning and decode using outfit. That give warnings only for the problematic strings.